test/gateway/eip20_cogateway/progress_mint.js

```diff
@@ -0,0 +1,398 @@
+    Undeclared: 0,
+    Declared: 1,
+    Progressed: 2,
+    DeclaredRevocation: 3,
+    Revoked: 4
+};
+
+async function _setup(accounts) {
+    
+    valueToken = accounts[0];
+    burner = accounts[10];
+    mockSafeCore = accounts[11];
+    membersManager = accounts[2];
+    coGateway = accounts[3];
+    owner = accounts[8];
+    testUtilityToken = await TestUtilityToken.new(
+        symbol,
+        name,
+        decimals,
+        valueToken,
+    );
+    bountyAmount = new BN(100);
+    staker = accounts[7];
+    stakerBalance = new BN(1000000);
+    rewardAmount = new BN(100);
+    
+}
+
+contract('EIP20CoGateway.progressMint() ', function (accounts) {
+    
+    let amount = new BN(200),
+        beneficiary = accounts[4],
+        gasPrice,
+        gasLimit,
+        nonce = new BN(1),
+        hashLockObj = Utils.generateHashLock(),
+        facilitator = accounts[5],
+        intentHash,
+        hashLock,
+        unlockSecret,
+        testEIP20CoGateway,
+        messageHash;
+    helper = new EIP20CoGatewayHelper();
+    
+    beforeEach(async function () {
+        
+        await _setup(accounts);
+        amount = new BN(200);
+        hashLock = hashLockObj.l;
+        unlockSecret = hashLockObj.s;
+        gasPrice = new BN(10);
+        gasLimit = new BN(10);
+        
+        intentHash = await helper.hashRedeemIntent(
+            amount,
+            beneficiary,
+            facilitator,
+            nonce,
+            gasPrice,
+            gasLimit,
+            valueToken,
+        );
+        testEIP20CoGateway = await TestEIP20CoGateway.new(
+            valueToken,
+            testUtilityToken.address,
+            mockSafeCore,
+            bountyAmount,
+            membersManager,
+            coGateway,
+            burner,
+        );
+        
+        await testUtilityToken.setCoGateway(testEIP20CoGateway.address);
+        await helper.setCoGateway(testEIP20CoGateway.address);
+        
+        messageHash = await testEIP20CoGateway.setStakeMessage.call(
+            intentHash,
+            nonce,
+            gasPrice,
+            gasLimit,
+            hashLock,
+            staker,
+        );
+        await testEIP20CoGateway.setStakeMessage(
+            intentHash,
+            nonce,
+            gasPrice,
+            gasLimit,
+            hashLock,
+            staker,
+        );
+    
+        await testEIP20CoGateway.setMints(messageHash, beneficiary, amount);
+        
+    });
+    
+    it('should mint for non-zero facilitator reward', async function () {
+        
+        await testEIP20CoGateway.setInboxStatus(
+            messageHash,
+            MessageStatusEnum.Declared,
+        );
+        
+        let progressMintValues = await testEIP20CoGateway.progressMint.call(
+            messageHash,
+            unlockSecret,
+            {from: facilitator},
+        );
+        
+        let expectedMintedToken = new BN(100),
+            expectedReward = new BN(100);
```

Comment:
How do you know `reward` is 💯  ? 🤔

Reply:
As discussed, calculation of reward will be done new ticket and this test-case will be updated.